mdtraj/core/topology.py

```diff
@@ -114,16 +121,15 @@ def _topology_from_subset(topology, atom_indices):
                     )
                     old_atom_to_new_atom[atom] = newAtom
 
-    bondsiter = topology.bonds
+    bondsiter: Iterable[Bond] | Callable[[], Iterable[Bond]] = topology.bonds
     if not hasattr(bondsiter, "__iter__"):
         bondsiter = bondsiter()
 
     for bond in bondsiter:
         try:
-            atom1, atom2 = bond
             newTopology.add_bond(
-                old_atom_to_new_atom[atom1],
-                old_atom_to_new_atom[atom2],
+                old_atom_to_new_atom[bond.atom1],
```

Comment:
Oops! Missed this one. Can you change this back to the old usage like we discussed?